MetaMask/metamask-extension
# Improve snap export metrics throttling

## **Description**
The previous throttling on the snap export metrics was too aggressive and would skip out on a bunch of events for some snaps. This PR improves this by changing the throttling logic to be per snap and per export.

## **Manual testing steps**

1. Install two different snaps (one using RPC and one using insights)
2. Make multiple RPC requests to one of the snaps and see that it is throttled
3. Create a transaction and view the insights tab multiple times, verify that the event for that is throttled
